torch/optim/adamw.py

```diff
@@ -424,15 +424,19 @@ def _single_tensor_adamw(
             if amsgrad:
                 # Maintains the maximum of all 2nd moment running avg. till now
                 if differentiable:
-                    max_exp_avg_sqs_i = max_exp_avg_sqs[i].clone()
+                    max_exp_avg_sq = max_exp_avg_sq.clone()
+                torch.maximum(max_exp_avg_sq, exp_avg_sq, out=max_exp_avg_sq)
+                
+                if torch.is_complex(max_exp_avg_sqs[i]):
+                    max_exp_avg_sqs[i] = torch.view_as_complex(max_exp_avg_sq)
                 else:
-                    max_exp_avg_sqs_i = max_exp_avg_sqs[i]
-                max_exp_avg_sqs[i].copy_(torch.maximum(max_exp_avg_sqs_i, exp_avg_sq))
+                    max_exp_avg_sqs[i] = max_exp_avg_sq
+
                 # Uses the max. for normalizing running avg. of gradient
                 # Folds in (admittedly ugly) 1-elem step_size math here to avoid extra param-set-sized read+write
                 # (can't fold it into addcdiv_ below because addcdiv_ requires value is a Number, not a Tensor)
                 denom = (
-                    max_exp_avg_sqs[i].sqrt() / (bias_correction2_sqrt * step_size_neg)
+                    max_exp_avg_sq.sqrt() / (bias_correction2_sqrt * step_size_neg)
```

Comment:
You want the complex version of this variable here?

Reply:
I want the one that’s been view_as_real’d